plugin/types.go

```diff
@@ -191,3 +192,24 @@ type Readable interface {
 	Entry
 	Open(context.Context) (SizedReader, error)
 }
+
+// ErrorFields represents the fields of an ErrorObj
+type ErrorFields = map[string]interface{}
+
+// ErrorObj represents a structured error object. If a plugin returns this type, it will be
+// passed back as part of an API response.
+type ErrorObj struct {
```

Comment:
Would a `NewErrorObj(Kind, Msg, Fields)` helper make sense? I'm asking b/c that would centralize the `puppetlabs.wash` prefix in all of our Kind fields.

Reply:
I'm not sure. The reason kind are constants are so you can match against them. We'd have to have to use a similar builder on the command side, which seems more complicated.

Reply:
Yeah, it's probably not that big of a deal anyways.